File: Rust/selection_sort.rs
//Example declaration in the main function should be let mut v = [..,..,..]; 
//and then function call --> selection_sort(&mut v);

fn selection_sort(array: &mut [i32]) {
    let mut min;
    for i in 0..array.len() {
        min = i;
        for j in (i+1)..array.len() {
            if array[j] < array[min] {
                min = j;
            }
        }
        let tmp = array[i];
        array[i] = array[min];
        array[min] = tmp;
    }
}

File: Go/selection_sort.go
package main

func selectionSort(items []int) []int {

	for i := 0; i < len(items); i++ {
		min := i
		for j := i + 1; j < len(items); j++ {
			if items[j] < items[min] {
				min = j
			}
		}

		tmp := items[i]
		items[i] = items[min]
		items[min] = tmp
	}
	return items
}
